plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/table/HudiTableFileSystemView.java

```diff
@@ -297,9 +296,7 @@ private FileIterator listPartition(Location partitionLocation)
         if (fileIterator.hasNext()) {
             return fileIterator;
         }
-        try (OutputStream ignored = metaClient.getFileSystem().newOutputFile(partitionLocation).create()) {
```

Comment:
I am not sure why this was getting created in the first place. Hudi connector does not support writes. In my opinion, the fix is valid, we just directly return empty iterator when partition oes not exist.
@anusudarsan @homar Do you recall any specific scenario why this was added?